app/queue.py

```diff
@@ -100,6 +104,11 @@ def poll(self, count=10) -> tuple[UUID, list[str]]:
         results = self.__move_to_inflight(in_flight_key, count)
         return (receipt, results)
 
+    def expire_inflights(self):
+        for key in self._redis_client.scan_iter(f"{Buffer.IN_FLIGHT.value}*"):
+            if self._redis_client.object("idletime", key) > self._expire_inflight_after_seconds:
+                self.move_from_inflight(key)
```

Comment:
I would have all of this logic within the LUA script. This will execute faster on the Redis side and be done within one sweep, without making network calls on every scans.